BUG: Unable to run garden tools command inside exec modules

## Bug

### Current Behavior
Here is what I am trying to do:

```
---
kind: Module
type: exec
name: tf-tasks
local: true
exclude: ['*']
tasks:
  - name: tf-workspace-init
    command: ["garden tools terraform-0-12-26 -- workspace new ${var.deployEnv} || garden tools terraform-0-12-26 -- workspace select ${var.deployEnv}"]
```

However I get the following error:

```
Failed running task tf-workspace-init in module tf-tasks. Here is the output:
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
internal/modules/cjs/loader.js:983
  throw err;
  ^
Error: Cannot find module '/Users/ejr/code/scratch/garden/demo-web-py/build/terraform/tools'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
    at Function.Module._resolveFilename (pkg/prelude/bootstrap.js:1346:46)
    at Function.Module._load (internal/modules/cjs/loader.js:862:27)
    at Function.Module.runMain (pkg/prelude/bootstrap.js:1375:12)
    at internal/main/run_main_module.js:18:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

moving the exec module into my root garden.yml causes a similar error...

### Expected behavior

I should be able to run the garden tools commands inside of an exec module command, to run out of band tasks for tools such as terraform, kubectl, etc... 

### Reproducible example

Any task running garden tools command

```
---
kind: Module
type: exec
name: tf-tasks
local: true
exclude: ['*']
tasks:
  - name: tf-workspace-init
    command: ["garden tools <tool> <args>"]
```

### Workaround

my specific use case was resolved by doing the following:

```
---
kind: Module
type: exec
name: tf-tasks
local: true
exclude: ['*']
tasks:
  - name: tf-workspace-set
    command: ["echo ${var.deployEnv} > .terraform/environment"]
```

and by passing the garden tools commands all together... 

### Suggested solution(s)

Not sure... 

### Additional context

Slack conversation link: https://kubernetes.slack.com/archives/CKM7CP8P9/p1598982704073300

### Your environment

* **OS:** macOS
* **How I'm running Kubernetes:**  EKS

* **Garden Version** 0.12.3

--- a/cli/package.json
+++ b/cli/package.json
@@ -38,7 +38,9 @@
     "minimist": "^1.2.5",
     "mocha": "^8.1.1",
     "mocha-logger": "^1.0.6",
-    "pkg": "4.4.9",
+    "patch-package": "^6.4.7",
+    "pkg": "4.5.1",
+    "postinstall-postinstall": "^2.1.0",
     "prettier": "^2.1.0",
     "split2": "^3.2.2",
     "tslint": "^6.1.3",
@@ -58,6 +60,7 @@
     "fix-format": "prettier --write \"{src,test}/**/*.ts\"",
     "generate-docs": "node ./build/src/generate-docs.js",
     "lint": "tslint -p .",
+    "postinstall": "patch-package",
     "test": "mocha"
   },
   "pkg": {
@@ -66,6 +69,11 @@
       "build/src/cli.js",
       "node_modules/@garden-io/**/*.js",
       "node_modules/es-get-iterator/**/*.js"
+    ]
+  },
+  "workspaces": {
+    "nohoist": [
+      "pkg"
     ]
   }
 }

--- a/cli/patches/pkg+4.5.1.patch
+++ b/cli/patches/pkg+4.5.1.patch
@@ -1,0 +1,13 @@
+diff --git a/node_modules/pkg/prelude/bootstrap.js b/node_modules/pkg/prelude/bootstrap.js
+index b87902f..4e94661 100644
+--- a/node_modules/pkg/prelude/bootstrap.js
++++ b/node_modules/pkg/prelude/bootstrap.js
+@@ -1512,7 +1512,7 @@ function payloadFileSync(pointer) {
+     }
+     var opts = args[pos];
+     if (!opts.env) opts.env = require('util')._extend({}, process.env);
+-    if (opts.env.PKG_EXECPATH === 'PKG_INVOKE_NODEJS') return;
++    if (opts.env.PKG_EXECPATH !== undefined) return;
+     opts.env.PKG_EXECPATH = EXECPATH;
+   }
+ 

--- a/cli/src/build-pkg.ts
+++ b/cli/src/build-pkg.ts
@@ -21,7 +21,7 @@ require("source-map-support").install()
 const repoRoot = resolve(GARDEN_CLI_ROOT, "..")
 const tmpDir = resolve(repoRoot, "tmp", "pkg")
 const tmpStaticDir = resolve(tmpDir, "static")
-const pkgPath = resolve(repoRoot, "node_modules", ".bin", "pkg")
+const pkgPath = resolve(repoRoot, "cli", "node_modules", ".bin", "pkg")
 const pkgFetchPath = resolve(repoRoot, "node_modules", ".bin", "pkg-fetch")
 const distPath = resolve(repoRoot, "dist")
 const sqliteBinFilename = "better_sqlite3.node"

--- a/core/src/util/util.ts
+++ b/core/src/util/util.ts
@@ -650,6 +650,11 @@ export async function runScript({
   script: string
   envVars?: PrimitiveMap
 }) {
+  envVars = envVars || {}
+
+  // Workaround for https://github.com/vercel/pkg/issues/897
+  envVars.PKG_EXECPATH = ""
+
   // Run the script, capturing any errors
   const proc = execa("bash", ["-s"], {
     all: true,

--- a/support/alpine-builder.Dockerfile
+++ b/support/alpine-builder.Dockerfile
@@ -27,7 +27,7 @@ ADD sdk /tmp/sdk
 WORKDIR /tmp/cli
 
 RUN apk add --no-cache python make gcc g++ --virtual .build-deps && \
-  yarn --production && \
+  yarn && \
   # Fix for error in this particular package
   rm -rf node_modules/es-get-iterator/test && \
   apk del .build-deps
@@ -36,6 +36,6 @@ ADD static /garden/static
 
 # Create the binary
 RUN mkdir -p /garden \
-  && ../pkg/node_modules/.bin/pkg --target node12-alpine-x64 . --output /garden/garden \
+  && node_modules/.bin/pkg --target node12-alpine-x64 . --output /garden/garden \
   && cp node_modules/better-sqlite3/build/Release/better_sqlite3.node /garden \
   && /garden/garden version

